Horns development

aadashti opened the pull request:
Greetings,

This pull request is for ```Horns```' developement, a script that will allow for you to import your Rhino Linux installation from a configuration script.

To test the project, you can run:
```
cargo run config-restore <PATH>
# Or
cargo build
target/debug/horns config-restore <PATH>
```
> [!TIP]
> An example JSON file is included at src/
---
## A Brief explanation regarding this program:

- ```main.rs``` to check the command and guide the user if the command was wrong
- ```todo.rs``` reads the JSON configurations part by part. Each part of this JSON file is a process that will be done. This file, later, will give these parts to the particular module based on the ID alongside passing some information.

> [!TIP]
> Each process has the following parts:
> - ```ID``` determines the current process between installing packages and running scripts.
> - With ```title```, a brief explanation can be provided for each.
> - ```detail``` holds the data. These data are either the name of packages or a script. (EDIT: P.S. for packages, the first item is dedicated to select the package manager: [packagemanager, package1, package2, ..., package n])
> - ```ask``` is a boolean. If true, the code asks users' confirmation and vice versa.

- ```install.rs``` calls ```rhino-pkg``` to install multiple packages one by one.
- ```script.rs``` runs entry scripts.

aadashti committed:
new file:   Cargo.lock
	new file:   Cargo.toml
	new file:   example.json
	modified:   rhino-deploy
	new file:   src/main.rs
	new file:   src/modules/install.rs
	new file:   src/todos.rs

aadashti committed:
modified:   Cargo.toml
	renamed:    example.json -> src/example.json
	modified:   src/modules/install.rs
	new file:   src/modules/script.rs
	modified:   src/todos.rs

aadashti committed:
modified:   Cargo.lock
	modified:   src/example.json
	modified:   src/modules/install.rs
	modified:   src/modules/script.rs

aadashti committed:
modified:   src/modules/install.rs
modified:   src/modules/script.rs

aadashti committed:
modified:   src/example.json

aadashti committed:
modified:   src/example.json
modified:   src/modules/install.rs
modified:   src/modules/script.rs

aadashti committed:
modified:   Cargo.lock
modified:   Cargo.toml
modified:   src/example.json
modified:   src/main.rs

aadashti committed:
modified:   src/modules/install.rs
modified:   src/modules/script.rs

aadashti committed:
Update main.rs

aadashti committed:
modified:   src/example.json
modified:   src/modules/install.rs
modified:   src/todos.rs

aadashti committed:
Update install.rs

ajstrongdev reviewed (changes requested):
With so much discussed on Discord I decided to post the full recap on our open ticket on our issue tracker.

https://github.com/rhino-linux/tracker/issues/20#issuecomment-3146446099

This pull request is going to need serious reworking to meet the full refined project scope.

aadashti commented:
> With so much discussed on Discord I decided to post the full recap on our open ticket on our issue tracker.
> 
> [rhino-linux/tracker#20 (comment)](https://github.com/rhino-linux/tracker/issues/20#issuecomment-3146446099)
> 
> This pull request is going to need serious reworking to meet the full refined project scope.

@ajstrongdev Thank you for providing the recap; it has been reviewed.

Edit: Quote reply included.

aadashti committed:
modified:   Cargo.lock
	modified:   Cargo.toml
	new file:   src/args/json_to_array.rs
	new file:   src/args/mod.rs
	new file:   src/args/terminal.rs
	modified:   src/main.rs
	modified:   src/modules/install.rs
	new file:   src/restore/check.rs
	new file:   src/restore/restore_file.rs
	new file:   src/restore/session.rs

aadashti committed:
modified:   src/args/json_to_array.rs
modified:   src/args/mod.rs
modified:   src/main.rs
modified:   src/restore/check.rs
new file:   src/restore/mod.rs
modified:   src/restore/restore_file.rs
modified:   src/restore/session.rs

aadashti committed:
new file:   configs/2025-08-09-23-26-07-github.com_aadashti_test_directory
modified:   src/restore/restore_file.rs
modified:   src/restore/session.rs

aadashti committed:
new file:   .vscode/settings.json
	deleted:    configs/2025-08-09-23-26-07-github.com_aadashti_test_directory
	modified:   src/args/mod.rs
	new file:   src/args/pm/flatpak.rs
	new file:   src/args/pm/mod.rs
	deleted:    src/modules/install.rs
	deleted:    src/modules/script.rs
	modified:   src/restore/session.rs

aadashti committed:
modified:   Cargo.lock
	modified:   Cargo.toml
	new file:   src/args/Packages.toml
	new file:   src/args/json_and_config.rs
	modified:   src/args/json_to_array.rs
	modified:   src/args/mod.rs
	new file:   src/args/packageConfig.rs
	deleted:    src/args/pm/flatpak.rs
	deleted:    src/args/pm/mod.rs
	new file:   src/configs/2025-11-06-01-48-57-github.com_aadashti_test_directory
	new file:   src/configs/2025-11-06-01-50-31-github.com_aadashti_test_directory
	new file:   src/configs/2025-11-06-01-54-40-github.com_aadashti_test_directory
	new file:   src/configs/2025-11-06-01-56-04-github.com_aadashti_test_directory
	deleted:    src/example.json
	modified:   src/restore/check.rs
	modified:   src/restore/restore_file.rs
	modified:   src/restore/session.rs

aadashti committed:
deleted:    src/configs/2025-11-06-01-48-57-github.com_aadashti_test_directory
	deleted:    src/configs/2025-11-06-01-50-31-github.com_aadashti_test_directory
	deleted:    src/configs/2025-11-06-01-54-40-github.com_aadashti_test_directory
	deleted:    src/configs/2025-11-06-01-56-04-github.com_aadashti_test_directory

aadashti committed:
modified:   src/args/json_to_array.rs
	new file:   src/configs/2025-11-06-02-04-36-github.com_aadashti_test_directory

aadashti committed:
modified:   src/args/mod.rs
	deleted:    src/args/packageConfig.rs
	new file:   src/configs/2025-11-06-02-06-15-github.com_aadashti_test_directory

aadashti committed:
deleted:    src/configs/2025-11-06-02-04-36-github.com_aadashti_test_directory
	deleted:    src/configs/2025-11-06-02-06-15-github.com_aadashti_test_directory
